WordPress/openverse-frontend
# Add playwright e2e tests

## Fixes


Related to #387 by @zackkrida

## Description


This PR sets up e2e testing using [Playwright](https://playwright.dev/docs/intro), replacing the previously added e2e stubs that used Cypress.

### Tests added:
1. the URL parameters are correctly converted into the search state [server rendered]:
- media type from search path `openverse/search/<image>?q=cat` is used to choose the 'Image' tab on the search page.
- query `q` (search term) parameter `openverse/search/?q=<cat>` is used as the value for search input.
- query parameters are converted into filters.  Opening `openverse/search/image?q=cat&license=cc0,by&aspect_ratio=tall` sets the two filters of type `license` checked (CC0 and BY), as well as one filter of type `aspect_ratio`.

2. When searching from the homepage, the client-rendered search page displays the following search state items correctly: query, media type, filter parameters.

To complete this PR, the following pieces need to be added:

- [x]  filters: selection from the sidebar/toggling using the tags

- [x] loading additional media using the "Load more" button.
- [x] search from the content pages (ie `About us`) works.
- [ ] clicking on a single image/audio displays a page with correct info filled in.

Based on [the comment below](https://github.com/WordPress/openverse-frontend/pull/394#issuecomment-957362406), we should create new tickets for the tests of the last two items.

## Testing Instructions

Run the app using `npm run dev`, and then run the tests using `npm run test:e2e`. To see the browser, you can add `--headed` flag to the `test:e2e` script.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
